indexer/src/main/java/bio/terra/tanagra/indexing/job/bigquery/WriteEntitySearchByAttributes.java

```diff
@@ -52,57 +53,69 @@ public JobStatus checkStatus() {
 
   @Override
   public void run(boolean isDryRun) {
-    List<Field> fields =
-        searchTable.getColumnSchemas().stream()
-            .map(
-                columnSchema ->
-                    Field.newBuilder(
-                            columnSchema.getColumnName(),
-                            BigQueryBeamUtils.fromDataType(columnSchema.getDataType()))
-                        .setMode(searchTable.includeNullValues() ? Mode.NULLABLE : Mode.REQUIRED)
-                        .build())
-            .toList();
-
-    // Build a clustering specification.
-    Clustering clustering =
-        Clustering.newBuilder().setFields(searchTable.getAttributeNames()).build();
-
-    // Create an empty table with this schema.
-    TableId destinationTable =
-        TableId.of(
-            indexerConfig.bigQuery.indexData.projectId,
-            indexerConfig.bigQuery.indexData.datasetId,
-            getOutputTableName());
-    googleBigQuery.createTableFromSchema(destinationTable, Schema.of(fields), clustering, isDryRun);
-
-    // Build the query to insert to the search table using a select from the main entity table.
+    // Create table definition & build the query to insert into the search table
+    // using a select from the main entity table.
+    List<Field> fields = new ArrayList<>();
     List<String> insertColumns = new ArrayList<>();
-    insertColumns.add(entity.getIdAttribute().getName());
-
     List<String> selectColumns = new ArrayList<>();
-    selectColumns.add(entity.getIdAttribute().getName());
-
     List<String> crossJoins = new ArrayList<>();
     List<String> whereClauses = new ArrayList<>();
+
     searchTable
-        .getAttributeNames()
+        .getColumnSchemas()
```

Comment:
changes: updates to how the index table is written based on whether all entityMain columns are added